src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1435,6 +1435,12 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, int position, S
                 return IsInBlock(accessor.Body, position);
             }
 
+            var propertyDeclaration = containingMember as PropertyDeclarationSyntax;
+            if (propertyDeclaration != null)
+            {
+                return isAnyExpressionContext && propertyDeclaration.ExpressionBody.IsKind(SyntaxKind.ArrowExpressionClause);
```

Comment:
Do we have to check isExpressionContext here?  It isn't done above for regular methods and property accessors, which leads me to believe that it's the callers job to check if they are in an expression or not.